### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by bader**
Improve pattern matchers:
- SYCL **must be a directory** and **must be located in the root directory**

Co-locate directory owned by the tools team and compiler tests.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,35 +1,31 @@
...
 # Use runtime team as the umbrella for most of the tests
-SYCL @intel/llvm-reviewers-runtime
+/SYCL/ @intel/llvm-reviewers-runtime
 
-# AOT compilation
-SYCL/AOT @intel/dpcpp-tools-reviewers
-
-#BFloat16 conversion
-SYCL/BFloat16 @AlexeySotkin @intel/dpcpp-tools-reviewers
+# Group algorithms
```
Comment (vladimirlaz) [inline .github/CODEOWNERS:10]: ordering of groups is unclear. Can we use alphabetical ordering based on comments?

Set the order
 BFloat16
 Compiler tests
 ESIMD
 Plugin interface for L0

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (bader) [reply .github/CODEOWNERS:10]: I grouped them by code owners:
- runtime
- plugins
- esimd
- tools

I'm not sure if using comments is a good idea - most of them just copy-paste directory name.
Do you want to use alphabetical order for directories?

Comment (vladimirlaz) [reply .github/CODEOWNERS:10]: Sorting by directories will break grouping by code owners.
I revoke the comment

**Review by vladimirlaz — DISMISSED** [review]

**Comment by bader** [comment]
@intel/dpcpp-tools-reviewers, SYCL :: Printf/double.cpp fails on Windows platform. Please, investigate.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,35 +1,31 @@
...
+/SYCL/GroupAlgorithm/ @Pennycook @intel/llvm-reviewers-runtime
+/SYCL/SubGroup/ @Pennycook @intel/llvm-reviewers-runtime
+/SYCL/SubGroupMask/ @Pennycook @intel/llvm-reviewers-runtime
+/SYCL/GroupLocalMemory/ @Pennycook @intel/llvm-reviewers-runtime
 
-# Device code split
-SYCL/DeviceCodeSplit @intel/dpcpp-tools-reviewers
+# Plugin interface for Level Zero
+/SYCL/Plugin/*level-zero* @intel/dpcpp-l0-pi-reviewers
+/SYCL/Plugin/*level_zero* @intel/dpcpp-l0-pi-reviewers
```
Comment (bader) [inline .github/CODEOWNERS]:
```suggestion
/SYCL/Plugin/*level[-_]zero* @intel/dpcpp-l0-pi-reviewers
```

**Comment by AGindinson** [comment]
> SYCL :: Printf/double.cpp fails on Windows platform. Please, investigate.

Created https://github.com/intel/llvm-test-suite/pull/735 to disable the failures.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,35 +1,31 @@
...
-# AOT compilation
-SYCL/AOT @intel/dpcpp-tools-reviewers
-
-#BFloat16 conversion
-SYCL/BFloat16 @AlexeySotkin @intel/dpcpp-tools-reviewers
+# Group algorithms
+/SYCL/GroupAlgorithm/ @Pennycook @intel/llvm-reviewers-runtime
+/SYCL/SubGroup/ @Pennycook @intel/llvm-reviewers-runtime
+/SYCL/SubGroupMask/ @Pennycook @intel/llvm-reviewers-runtime
+/SYCL/GroupLocalMemory/ @Pennycook @intel/llvm-reviewers-runtime
```
Comment (bader) [inline .github/CODEOWNERS]:
```suggestion
/SYCL/GroupAlgorithm/ @Pennycook @intel/llvm-reviewers-runtime
/SYCL/GroupLocalMemory/ @Pennycook @intel/llvm-reviewers-runtime
/SYCL/SubGroup/ @Pennycook @intel/llvm-reviewers-runtime
/SYCL/SubGroupMask/ @Pennycook @intel/llvm-reviewers-runtime
```

**Comment by bader** [comment]
> > SYCL :: Printf/double.cpp fails on Windows platform. Please, investigate.
> 
> Created #735 to disable the failures.

Great. Thank you a lot!

File: .github/CODEOWNERS (hunk 1 above)
Comment (bader) [inline .github/CODEOWNERS:10]:
```suggestion

# SYCL sub-directory matchers are grouped by code owner first, followed by
# alphabetical order within the group. Please, keep this ordering.

# Group algorithms
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -1,35 +1,31 @@
...
-SYCL/GroupAlgorithm @Pennycook @intel/llvm-reviewers-runtime
-SYCL/SubGroup @Pennycook @intel/llvm-reviewers-runtime
-SYCL/SubGroupMask @Pennycook @intel/llvm-reviewers-runtime
-SYCL/GroupLocalMemory @Pennycook @intel/llvm-reviewers-runtime
+/SYCL/ESIMD/ @intel/dpcpp-esimd-reviewers
 
-# Plugin interface for Level Zero
-SYCL/Plugin/*level-zero* @intel/dpcpp-l0-pi-reviewers
-SYCL/Plugin/*level_zero* @intel/dpcpp-l0-pi-reviewers
+# Compiler tests
```
Comment (bader) [inline .github/CODEOWNERS:25]:
```suggestion
# BFloat16 conversion
/SYCL/BFloat16/ @AlexeySotkin @intel/dpcpp-tools-reviewers

# Compiler tests
```

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -1,35 +1,31 @@
...
-# Separate compilation
-SYCL/SeparateCompile @intel/dpcpp-tools-reviewers
+/SYCL/Printf/ @intel/dpcpp-tools-reviewers
 
 # Specialization constant
-SYCL/SpecConstants @intel/dpcpp-tools-reviewers
+/SYCL/SpecConstants/ @intel/dpcpp-tools-reviewers
+
+# BFloat16 conversion
+/SYCL/BFloat16/ @AlexeySotkin @intel/dpcpp-tools-reviewers
```
Comment (bader) [inline .github/CODEOWNERS:31]:
```suggestion
/SYCL/SpecConstants/ @intel/dpcpp-tools-reviewers
```

File: .github/CODEOWNERS (hunk 1 above)
Comment (bader) [reply .github/CODEOWNERS:10]: I've added a comment defining the grouping and sorted sub-directories alphabetically within the groups.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**